### PR TITLE
The tooltips your mod wishes it had

### PR DESCRIPTION
**Pull request by YannickMG**
See GTNewHorizons/GT-New-Horizons-Modpack#9367 for details.

Also added 2 configurable levels of verbosity, the second of which applies when LSHIFT is held down. This replaces the proposed "tooltip toggle" button from the RFC.

Special thanks to @Glease for enabling this feature creep.

### Verbosity and LSHIFT aware tooltips in action
shift-tooltips

### Verbosity config options
image

### Bonus
lol-voltage

### CODE REVIEW

**Comment by YannickMG** [comment]
I should of course note that the text added to the language file is completely open to whatever changes are deemed fit, and in most cases you shouldn't need to touch the code to change the different verbosity levels of each tooltip unless you want to add new formatting arguments. It's intended to make it really easy to add more tooltips to the rest of the UI wherever needed.

Here's the current reasoning behind the implemented verbosity levels:
- 0 : Some people just don't want to see tooltips, or they only want to see them when holding down LSHIFT
- 1 : Let's you easily figure out what a thing is and opt in to see more details with LSHIFT without flooding you with info.
- 2 : Is intended to have either critical info that a noob needs and an experienced player would not even look at anyway, or to have genuinely useful information for all skill levels.
- 3 : Is intended to have extended tips and information for new players learning the pack.

**Comment by YannickMG** [comment]
Note that there will probably be some merge conflicts for me to resolve on either PR once this or #856 is merged as both PRs add config options  and mess with metatiles and basic machines.

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_BasicMachine_Bronze.java [hunk 1]
```diff
@@ -38,6 +39,7 @@ public IMetaTileEntity newMetaEntity(IGregTechTileEntity aTileEntity) {
         return new GT_MetaTileEntity_BasicMachine_Bronze(mTier, mDescription, mTextures);
     }
     */
+    private static final int NEEDS_STEAM_VENTING = 64;
```
Comment (repo-alt) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_BasicMachine_Bronze.java:42]: A constant defined in 2 places?

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_BasicMachine_Bronze.java [hunk 2]
```diff
@@ -150,6 +152,13 @@ public boolean allowToCheckRecipe() {
         return !mNeedsSteamVenting;
     }
 
+    @Override
+    public void onPostTick(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {
+        super.onPostTick(aBaseMetaTileEntity, aTick);
+        // Super already zeroed out setErrorDisplayID, add additional error codes here.
+        aBaseMetaTileEntity.setErrorDisplayID(aBaseMetaTileEntity.getErrorDisplayID() | (mNeedsSteamVenting ? 64 : 0));
```
Comment (repo-alt) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_BasicMachine_Bronze.java:159]: (and not used after all)

**Review by repo-alt — APPROVED** [review]
In general it looks ok, but the PR is quite big so maybe someone else may look at that

**Comment by Dream-Master** [comment]
Let’s zeta test this stuff

**Pull request merged**